torch/csrc/jit/codegen/cuda/evaluator_common.cpp

```diff
@@ -434,22 +434,34 @@ void KernelPrecomputedIntegers::initializeNamedScalars() {
   }
 }
 
+// TODO: merge this one with above.
 void KernelPrecomputedIntegers::bindKernelInputs(
     kir::Kernel* kernel,
-    const at::ArrayRef<IValue>& aten_inputs) {
+    const KernelArgumentHolder& args) {
   if (hasValidValues()) {
     invalidate();
   }
 
   const auto& inputs = kernel->inputs();
 
   for (const auto i : c10::irange(inputs.size())) {
+    auto arg = args[i];
     const auto input = inputs[i];
     if (auto tensor_input = dynamic_cast<TensorView*>(input)) {
-      const auto aten_tensor = aten_inputs[i].toTensor();
-      bindTensorMetaData(tensor_input, aten_tensor);
+      if (const auto& tensor_arg_abstract =
+              dynamic_cast<const TensorArgAbstract*>(arg)) {
+        bindTensorMetaData(tensor_input, tensor_arg_abstract);
+      } else {
+        TORCH_CHECK(
+            arg->isType(ArgType::CpuScalarTensor),
+            "binding input to TensorView expects input arg to be of tensor type");
+      }
     } else if (input->isScalar() && input->dtype() == DataType::Int) {
```

Comment:
hmmm.... That's a good question.
I think on paper it should be yes. because people can use cpu scalar in similar manner. Putting a TODO there
```
In [13]: v = torch.tensor(3)

In [14]: x = torch.randn(15, device="cuda")

In [15]: x.reshape(v, -1)
Out[15]: 
tensor([[ 0.7362,  0.1321,  0.0773,  0.1388,  1.0230],
        [ 1.0304, -0.5529, -0.2022, -0.2125,  1.1802],
        [ 0.2915, -0.6966, -0.8211,  0.6842, -0.5666]], device='cuda:0')
```